sentry-ruby/CHANGELOG.md

```diff
@@ -18,6 +18,11 @@ The SDK now records a new `net.http` breadcrumb whenever the user makes a reques
 
 <img width="869" alt="net http breadcrumb" src="https://user-images.githubusercontent.com/5079556/114298326-5f7c3d80-9ae8-11eb-9108-222384a7f1a2.png">
 
+- Add the third tracing state [#1402](https://github.com/getsentry/sentry-ruby/pull/1402)
+  - `rate == 0` - Tracing enabled. Rejects all locally created transactions but  respects sentry-trace.
+  - `1 > rate > 0` - Tracing enabled. Samples locally created transactions  with the rate and respects sentry-trace.
+  - `rate < 0` or `rate > 1` - Tracine disabled.
```

Comment:
This PR doesn't reference anything, so I don't know what was the motivation behind the change.

I can point to https://github.com/getsentry/develop/issues/265

What we've been converging towards when clearing up inconsistencies is that `traces_sample_rate` > 0 or `traces_sampler != nil` enables tracing.

`traces_sample_rate` not explicitly set, `nil`, means no transactions are sent. The `sentry-trace` header is only looked into if tracing is enabled as per definition above.

https://develop.sentry.dev/sdk/performance/#tracessamplerate `traces_sample_rate` out of the [0, 1] range is undefined behavior.

Reply:
sorry I should've added some references here, but I thought it's already a consensus. here are the 2 drivers:
- I saw `sentry-python` defined it as [invalid](https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/tracing.py#L698-L704) 
- @lobsterkatie also wrote [a test case](https://github.com/getsentry/sentry-ruby/pull/1328/files#diff-ca2b59d3ec608f0ce8907b919472b48f2b603b28343985143e9f1ef364b42c20R36-R43) to declare that `rate > 1` is invalid and should disable tracing.

Reply:
```suggestion
  - `rate < 0` or `rate > 1` - Tracing disabled.
```